### PR TITLE
Chore: Refactor to use messageId in a number of rules

### PR DESCRIPTION
**Pull request by bradzacher**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Other: Helping to bring the codebase closer to standardisation.

**What changes did you make? (Give an overview)**

I went through the rules and added `messages` to a number of older rules.
I picked non-deprecated rules, and rules that could be simply changed (i.e. essentially just a prop change in `context.report`).

I did as many as I could before getting bored 😅 (there's about 71 more rules to go)

There should be no functionality changes at all from this PR.
No new tests have been added, and no tests have been changed (apart from switching from hardcoded `message` to `messageId` + `data`).
Whilst I was at it, I did a little bit of reformatting to standardise on the format style I saw in other rules.

Let me know if you want me to split this up into separate PRs because of its size.

### CODE REVIEW

File: lib/rules/no-invalid-regexp.js [hunk 1]
```diff
@@ -40,6 +40,8 @@ module.exports = {
             },
             additionalProperties: false
         }]
+
+        // no messages, because the error text comes directly from the regexpp module
```
Comment (kaicataldo) [inline lib/rules/no-invalid-regexp.js]: :+1: Thanks for this helpful comment

File: tests/lib/rules/no-restricted-globals.js [hunk 2]
```diff
@@ -60,122 +60,210 @@ ruleTester.run("no-restricted-globals", rule, {
...
+            errors: [{
+                messageId: "defaultMessage",
+                data: { name: "foo" },
+                type: "Identifier"
+            }]
         },
         {
             code: "foo",
             options: [{ name: "foo", message: "Use bar instead." }],
-            errors: [{ message: "Unexpected use of 'foo'. Use bar instead.", type: "Identifier" }]
```
Comment (kaicataldo) [inline tests/lib/rules/no-restricted-globals.js]: Could be nice to assign `"Use bar instead"` to a variable we can reuse in this file.

**Comment by G-Rath** [comment]
> I did as many as I could before getting bored 😅 (there's about 71 more rules to go)

@kaicataldo @ilyavolodin I don't mind tackling the rest of them (or at least until I've gotten board as well), as it would give me a good excuse to ~rummage around~ carefully & respectively explore the codebase.

Could be a good bullet point for the v7 release if they all got done in time?

**Comment by kaicataldo** [comment]
Great! Let us know if we can help with anything.

> Could be a good bullet point for the v7 release if they all got done in time?

I don’t see why not! Any PRs that are merged are always listed in the release notes blog post as well :)

**Pull request merged**